geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ServerConnectionCollection.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.geode.internal.cache.tier.sockets;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.concurrent.atomic.AtomicInteger;
+
+// This is used to form of group of connections for a particular client.
+public class ServerConnectionCollection {
+  private Set<ServerConnection> connectionSet = new HashSet<ServerConnection>();
+
+  // Number of connections currently processing messages for this client
+  final AtomicInteger connectionsProcessing = new AtomicInteger();
+
+  // Indicates that the server is soon to be or already in the process of terminating connections in
+  // this collection.
+  boolean isTerminating = false;
```

Comment:
Should this be volatile or maybe an AtomicBoolean?  It's accessed under synchronization on different objects.  CHM uses its proxyidcollections object and ServerConnection is using its serverConnectionCollection.

Reply:
Good catch, making this volatile.